packages/vhd-lib/Vhd/VhdSynthetic.js

```diff
@@ -40,13 +43,38 @@ exports.VhdSynthetic = class VhdSynthetic extends VhdAbstract {
     }
   }
 
-  static async open(vhds) {
-    const vhd = new VhdSynthetic(vhds)
+  static async fromVhdChain(handler, childPath) {
+    const disposableVhds = []
+    let vhd
+    let vhdPath = childPath
+    do {
+      const disposable = await openVhd(handler, vhdPath)
```

Comment:
This is a bad idea, what happens if one of the VHDs fails to open?

This goes against the principle of disposables: to guarantee that all disposables are disposed of, no matter what happens.

Maybe [`DIsposable.factory`](https://github.com/JsCommunity/promise-toolbox#creation-1) would help in this case. :slightly_smiling_face:

Reply:
Can you explain how to use Disposable.factory ?